android/modules/ui/src/java/ti/modules/titanium/ui/widget/TiUILabel.java

```diff
@@ -51,8 +56,12 @@
 	private float shadowX = 0f;
 	private float shadowY = 0f;
 	private int shadowColor = Color.TRANSPARENT;
-	private String minimumFontSize = null;
-	private String autoshrinkSetFontSize = null;
+	private int autoLinkFlags;
+	private int viewHeightInLines;
+	private int maxLines = Integer.MAX_VALUE;
+	private float minimumFontSizeInPixels = -1.0f;
```

Comment:
1. Good point, but currently minimumFontSize only accepts a `Number`. You cannot specify units like [fontSize](http://docs.appcelerator.com/platform/latest/#!/api/Font-property-fontSize) which we need to fix.

You can make use of these methods, like we do when setting `fontSize`
```
// TiUIHelper.java - org.appcelerator.titanium.util;
getSizeUnits(fontSize)
getSize(fontSize)

tv.setTextSize(getSizeUnits(fontSize), getSize(fontSize))
```

2. Just checked, we do use `float` values for defining `fontSize` but we use `int` for pixels in our LayoutEngine. Since you were converting to raw pixels I assumed using `int` would suffice and prevent rounding errors. But I agree using `float` is the better solution. I feel we shouldn't need to take multiple attempts to set the correct size though.